src/swupd/files_test.go

```diff
@@ -0,0 +1,178 @@
+package swupd
+
+import "testing"
+
+func TestSetFileTypeFromFlagFile(t *testing.T) {
+	flags := map[string]string{
+		"F": "file",
+		"D": "directory",
+		"L": "link",
+		"M": "manifest",
+		".": "",
+	}
+
+	for flag, ftype := range flags {
+		t.Run(flag, func(t *testing.T) {
+			f := File{}
+			if err := setFileTypeFromFlag(flag, &f); err != nil {
+				t.Errorf("failed to set %v type flag on file", flag)
+			}
+
+			if f.Type != ftype {
+				t.Errorf("file type was set to %v from %v flag", f.Type, flag)
+			}
+		})
+	}
+
+	// space is never valid
+	t.Run("\" \"", func(t *testing.T) {
```

Comment:
Go has more than one type of quote, so you can usually use a different one rather than needing to escae character.